### PR TITLE
Fix issues from mock pe

### PR DESCRIPTION
**Pull request by houtenteo**

### CODE REVIEW

**Review by Kangxun — COMMENTED** [review]
looks good just some minor edits 👍🏻

File: docs/UserGuide.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ If you can type fast, CMM can get your contact management tasks done faster than
    - `list` : Lists all contacts.
    - `done 2` : Marks the 2nd contact shown in the current list as completed.
    - `edit 1 n/Bob p/62353535` : Edits the name and phone number of the first person in the displayed list.
+   - `findAny n/alex g/m` : Finds the contacts that have the substring `alex`in their name **AND/OR** are males.
```
[the remark this answers is not in the document]
Comment (houtenteo) [reply docs/UserGuide.md:40]: wym by missing space?

File: src/main/java/seedu/address/logic/parser/FindAllCommandParser.java [hunk 2]
```diff
@@ -195,4 +196,21 @@ private void checkTrueOrFalse(String s) throws ParseException {
         }
     }
 
+    /**
+     * Checks if the string supplied is either "m", "male", "f", "female" or "n.a". (case-insensitive)
+     * @param s The String to be checked
+     * @return True if the string supplied is either "t", "true", "f" or "false". False otherwise
```
[the remark this answers is not in the document]
Comment (houtenteo) [reply src/main/java/seedu/address/logic/parser/FindAllCommandParser.java]: this was fixed in later commits

File: src/main/java/seedu/address/logic/parser/FindAnyCommandParser.java [hunk 3]
```diff
@@ -193,4 +194,21 @@ private void checkTrueOrFalse(String s) throws ParseException {
         }
     }
 
+    /**
+     * Checks if the string supplied is either "m", "male", "f", "female" or "n.a". (case-insensitive)
+     * @param s The String to be checked
+     * @return True if the string supplied is either "t", "true", "f" or "false". False otherwise
```
[the remark this answers is not in the document]
Comment (houtenteo) [reply src/main/java/seedu/address/logic/parser/FindAnyCommandParser.java]: fix in later commits as well

File: src/main/java/seedu/address/model/person/Phone.java [hunk 4]
```diff
@@ -11,8 +11,8 @@
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Phone numbers should only contain numbers, and it should be at least 3 digits long";
-    public static final String VALIDATION_REGEX = "\\d{3,}";
+            "Phone numbers should only contain numbers, and it should be at least 8 digits long";
```
[the remark this answers is not in the document]
Comment (houtenteo) [reply src/main/java/seedu/address/model/person/Phone.java]: ok, updated the regex

File: docs/UserGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kangxun) [reply docs/UserGuide.md:40]: *missing space after `alex`

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -241,4 +263,142 @@ public static Category parseCategory(String category) throws ParseException {
...
+     * @return The predicate
+     */
+    public static InterestContainsKeywordsPredicate getInterestPredicate(String interestKeywords, boolean isFindAll) {
+        String[] interestKeywordsArr = interestKeywords.split("\\s+");
+        return new InterestContainsKeywordsPredicate(Arrays.asList(interestKeywordsArr), isFindAll);
+    }
+
+    /**
+     * Checks if the string supplied is either "t", "true", "f" or "false". (case-insensitive)
+     * @param s The String to be checked
```
Comment (Kangxun) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:365]: maybe add @throws

File: docs/UserGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (houtenteo) [reply docs/UserGuide.md:40]: ah okok

**Review by Kangxun — APPROVED** [review]
lgtm

**Pull request merged**